craft_parts/executor/part_handler.py

```diff
@@ -340,12 +341,25 @@ def _run_step(
         :return: If step is Stage or Prime, return a tuple of sets containing
             the step's file and directory artifacts.
         """
+        step_env = generate_step_environment(
+            part=self._part, plugin=self._plugin, step_info=step_info
+        )
+
+        # Validate build environment
```

Comment:
add comment about differences with the one at the executor level

Reply:
Added clarification.